src/platform/ESP32/DiagnosticDataProviderImpl.cpp

```diff
@@ -125,6 +130,12 @@ CHIP_ERROR DiagnosticDataProviderImpl::GetCurrentHeapHighWatermark(uint64_t & cu
     return CHIP_NO_ERROR;
 }
 
+CHIP_ERROR DiagnosticDataProviderImpl::ResetWatermarks()
+{
+    // esp-idf does not provide a way to reset the watermarks
+    return CHIP_ERROR_NOT_IMPLEMENTED;
```

Comment:
The spec does not allow having the watermark feature but not having a working ResetWatermarks command....  I would expect devices that have this code to fail certification as things stand: see https://github.com/CHIP-Specifications/chip-test-plans/blob/master/src/cluster/software_diagnostics.adoc#513-tc-dgsw-23-command-received-dut_sever step 5.

Reply:
Thanks for the test-plan pointer, let me re-check what can be done.